x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/session/EqlSession.java

```diff
@@ -116,14 +120,25 @@ private <T> void preAnalyze(LogicalPlan parsed, ActionListener<LogicalPlan> list
             listener.onFailure(new TaskCancelledException("cancelled"));
             return;
         }
+        Set<String> fieldNames = fieldNames(parsed);
         indexResolver.resolveAsMergedMapping(
             indexWildcard,
+            fieldNames,
             configuration.indicesOptions(),
             configuration.runtimeMappings(),
             map(listener, r -> preAnalyzer.preAnalyze(parsed, r))
         );
     }
 
+    static Set<String> fieldNames(LogicalPlan parsed) {
+        Set<String> fieldNames = new HashSet<>();
```

Comment:
Using a LinkedHashMap makes debugging easier.

Reply:
I assumed you meant `LinkedHashSet`.